core/src/test/scala/unit/kafka/server/epoch/LeaderEpochIntegrationTest.scala

```diff
@@ -142,17 +143,20 @@ class LeaderEpochIntegrationTest extends QuorumTestHarness with Logging {
     fetcher1.close()
   }
 
-  @Test
-  def shouldIncreaseLeaderEpochBetweenLeaderRestarts(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def shouldIncreaseLeaderEpochBetweenLeaderRestarts(quorum: String): Unit = {
     //Setup: we are only interested in the single partition on broker 101
-    brokers += createServer(fromProps(createBrokerConfig(100, zkConnect)))
-    assertEquals(100, TestUtils.waitUntilControllerElected(zkClient))
+    brokers += createBroker(fromProps(createBrokerConfig(100, zkConnectOrNull)))
+    if (!isKRaftTest()) {
```

Comment:
Can we have a similar assertion in KRaft mode?

Reply:
I checked the leader election using `KafkaRaftClient` in KRaft Mode. Is this a good approach?